src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,239 @@
+import fs from 'fs'
+import * as simpleGit from 'simple-git'
+import {Writable} from 'stream'
+import {URL} from 'url'
+import {promisify} from 'util'
+
+import {renderGitError, renderProjectPathNotFoundWarning, renderSourcesNotFoundWarning} from './renderer'
+
+// NewSimpleGit returns a configured SimpleGit.
+export const newSimpleGit = (): simpleGit.SimpleGit => {
+  const options: simpleGit.SimpleGitOptions = {
+    baseDir: process.cwd(),
+    binary: 'git',
+    // We are invoking at most 3 git commands at the same time.
+    maxConcurrentProcesses: 3,
+  }
+
+  return simpleGit.gitP(options)
+}
+
+// GitRemote returns the remote of the current repository.
+const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
+  const remotes = await git.getRemotes(true)
+  if (remotes.length === 0) {
+    throw new Error('No git remotes available')
+  }
+  remotes.forEach((remote) => {
+    if (remote.name === 'origin') {
+      return remote.refs.push
+    }
+  })
+
+  return remotes[0].refs.push
+}
+
+// StripCredentials removes credentials from a remote HTTP url.
+export const stripCredentials = (remote: string) => {
+  try {
+    const url = new URL(remote)
+    url.username = ''
+    url.password = ''
+
+    return url.toString()
+  } catch {
+    return remote
+  }
+}
+
+// GitHash returns the hash of the current repository.
+const gitHash = async (git: simpleGit.SimpleGit): Promise<string> => git.revparse('HEAD')
+
+// GitTrackedFiles returns the tracked files of the current repository.
+export const gitTrackedFiles = async (git: simpleGit.SimpleGit): Promise<string[]> => {
+  const files = await git.raw('ls-files')
+
+  return files.split(/\r\n|\r|\n/)
+}
+
+// TrimStart trims from a set of characters from the start of a string.
```

Comment:
NIT: This was confusing for the following function. No need for the name of the function here.
The same comment for other functions as well, or at least follow the same casing as the function name.

```suggestion
// trims from a set of characters from the start of a string.
```

Reply:
I could not keep the casing because of linting errors `comment must start with uppercase letter`. I'll adjust the comments 👍